doc/src/guides/solving/solve-diophantine-equation.md

```diff
@@ -0,0 +1,207 @@
+
+### Expression to Solve Must Equal Zero
+
+{func}`~.diophantine` requires that the expression to be solved equals zero, so
+you must transform an equation into such an expression. For example, $a^2 + b^2
+= c^2$ must be expressed as $a^2 + b^2 - c^2$ as in
+[](#example-of-solving-a-diophantine-equation).
+
+### Specify the Order of Symbols in the Result
+
+We recommend you specify the order of symbols in the result to avoid confusion.
+Use the `syms` parameter and pass it a tuple (or list) of symbols to ensure the
+result will be in that order.
+
+### Limitations
+
+Currently, following five types of Diophantine equations can be solved using
+{meth}`~sympy.solvers.diophantine.diophantine.diophantine` and other helper
+functions of the Diophantine module.
+
+- Linear Diophantine equations: $a_1x_1 + a_2x_2 + \ldots + a_nx_n = b$
+- General binary quadratic equation: $ax^2 + bxy + cy^2 + dx + ey + f = 0$
+- Homogeneous ternary quadratic equation: $ax^2 + by^2 + cz^2 + dxy + eyz + fzx
+  = 0$
+- Extended Pythagorean equation: $a_{1}x_{1}^2 + a_{2}x_{2}^2 + \ldots +
+  a_{n}x_{n}^2 = a_{n+1}x_{n+1}^2$
+- General sum of squares: $x_{1}^2 + x_{2}^2 + \ldots + x_{n}^2 = k$
+
+## Use the Solution Result
+
+### Extract Expressions From the Result
+
+{func}`~.diophantine` returns results as a set of tuples, where each element in
+a tuple is an expression for a variable in your equation. For example, in
+
+```py
+>>> from sympy.solvers.diophantine import diophantine
+>>> from sympy import symbols
+>>> a, b, c, p, q = symbols("a, b, c, p, q", integer=True)
+>>> my_syms = (a, b, c)
+>>> d = diophantine(pythag, syms=my_syms)
+>>> d
+{(2*p*q, p**2 - q**2, p**2 + q**2)}
+```
+
+the result is a set containing one tuple where the expressions correspond to (a,
+b, c). That is, the tuple represents `a = 2*p*q, b = p**2 - q**2, c =
+p**2-q**2`.
+
+Because you cannot extract an element (here, a tuple) from a set by subscripting
+the set, you can convert the set to a list, and then subscript the list:
+
+```py
+>>> solution_list = list(d)
+>>> solution_list
+[(2*p*q, p**2 - q**2, p**2 + q**2)]
+>>> solution_list[0] # Extract a tuple corresponding to a solution
+(2*p*q, p**2 - q**2, p**2 + q**2)
+>>> solution_list[0][0] # Extract an expression for one variable, here a
+2*p*q
+```
+
+You can also create a dictionary of symbol-expression pairs to extract an
+expression by its symbol:
+
+```py
+>>> from sympy.solvers.diophantine import diophantine
+>>> from sympy import symbols
+>>> a, b, c = symbols("a, b, c", integer=True)
+>>> my_syms = (a, b, c)
+>>> solution, = diophantine(pythag, syms=my_syms)
+>>> solution
+(2*p*q, p**2 - q**2, p**2 + q**2)
+>>> solution_dict = dict(zip(my_syms, solution))
+>>> solution_dict
+{a: 2*p*q, b: p**2 - q**2, c: p**2 + q**2}
+>>> solution_dict[a]
+2*p*q
+```
+
+### Work With Parameters
+
+You can manipulate parameters such as `p` and `q`, which are generated
+automatically by {func}`~.diophantine`, by creating them as symbols. For
```

Comment:
It's important to include the `integer=True` assumption when doing this. Otherwise, it won't work.

Reply:
Conversely, specifying `integer=True` for the symbols that are part of the original equation is not required, although it is a good practice.